hack/scripts/async-delete-test.sh

```diff
@@ -0,0 +1,49 @@
+echo "create busybox deployment and verify async delete"
+kubectl apply -f ../manifests/busybox.yaml
+kubectl rollout status deployment busybox
+
+echo "temporarily disable CNS daemonset and attempt busybox pod delete"
+kubectl -n kube-system patch daemonset azure-cns -p '{"spec": {"template": {"spec": {"nodeSelector": {"non-existing": "true"}}}}}'
+
+echo "delete busybox pod"
+for node in $(kubectl get nodes -o name);
+do
+    node_name="${node##*/}"
+    busybox_pod=$(kubectl get pods -l k8s-app=busybox -o wide | grep "$node_name" | awk '{print $1}')
+    if [ -z $busybox_pod  ]; then
+        continue
+    else
+        echo "wait 1 min for delete to processes and error to catch. expect a file to be written to var/run/azure-vnet/deleteIDs"
+        kubectl delete pod $busybox_pod 
+        sleep 60s
+        
+        echo "restore azure-cns pods"
+        kubectl -n kube-system patch daemonset azure-cns --type json -p='[{"op": "remove", "path": "/spec/template/spec/nodeSelector/non-existing"}]'
```

Comment:
If the test fails to delete the pod does CNS get left in an unready state? Would like to be able to run the test without having to manually remove the patch.

Reply:
updated the e2e step with a check to remove the patch if it still exists